grafanalib/core.py

```diff
@@ -266,6 +398,13 @@ def to_json_data(self):
             'value_type': self.valueType,
         }
 
+    @staticmethod
+    def parse_json_data(data):
+        if 'value_type' in data:
+            data['valueType'] = data.pop('value_type')
+
+        return Tooltip(**data)
```

Comment:
Using this as a concrete example for some of the points mentioned in the top-level comment:

```python
@classmethod
def parse_json_data(cls, data):
    return cls(**rename_keys(data, value_type='valueType'))

def rename_keys(data, **keys):
    new_data = dict(data)
    for old_key, new_key in keys.items():
        if old_key in new_data:
            new_data[new_key] = new_data.pop(old_key)
    return new_data
```